### PR TITLE
API Foundation

### PR DESCRIPTION
**Pull request by dee-me-tree-or-love**
*What*
Mocking foundation of the `magic-mirror` service  

*Why* 
To start somewhere

*Usage*
Has a discover route, where with the body like `{"entities":["events"]}` that specifies the domains to be queried for data
_Notice plural `events`_

*Architecture*  
Built in Express. The components of the application are the following: 

:zap: `Router` 
to configure the application routes

:zap: `DiscoverHandler` 
the handler of the `/discover/` route. 
kinda like a controller, but named differently

:zap: `DomainCatalog` 
the catalog of known domains and their resolvers
think of a resolver factory of a sort
is setup upon a call to a static `getCatalog()` method

:zap: `IResolverBase` & Implementations  
the proxy that knows how to get the data of its domain
defines the query logic in `resolve()` method that returns a `Promise<Array>` 
knows own domain key and returns it with the method `getKey()` which returns a `String`

*Processes*
Basically when the discover route is called, the handler gets the `Resolver`s from the `DomainCatalog`.
Then it prepares the jobs for resolving the data. Then these jobs are asynchronously executed and the aggregate response is returned the the caller. 
See here for the flow at [`handle()`](https://github.com/Volunteero/magic-mirror/blob/304091fc7c598f5b99c3a40deb0f9cad1b7870a2/src/handlers/DiscoverHandler.js#L25)

### CODE REVIEW

**Comment by dee-me-tree-or-love** [comment]
For now it is s hard coded mock

image

**Comment by dee-me-tree-or-love** [comment]
oh, yeah, and it's deployed now to heroku
[`https://volunteero-magic-mirror.herokuapp.com/`](https://volunteero-magic-mirror.herokuapp.com/) 
:building_construction:

File: src/routes/Router.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    });
+
+    routes.push({
+      path: `get: ${heartbeatPath}`,
+    });
+
+
+    /**
+     * The handlers for specific endpoints go here
+     * TODO: maybe we could use a config file like a sls.yml but our own?
```
Comment (dpapukchiev) [inline src/routes/Router.js:49]: maybe dotenv npm package might be usefull in combination with a js file like this one
https://github.com/studyportals/StudentEvents/blob/develop/config/serverless/serverless.config.js

**Review by dpapukchiev — APPROVED** [review]

File: src/routes/Router.js (hunk 1 above)
Comment (dee-me-tree-or-love) [reply src/routes/Router.js:49]: Oh, that's awesome! Yeah, we could think of different options, but indeed could be useful. 
And then we could have the config first routing maybe (question is whether we even need and want it?)
